src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+        final File assembledContigsFile = new File(workingDir, contigsFileName);
+        return new SGAAssemblyResult(assembledContigsFile, runtimeInfo);
+    }
+
+    /**
+     * Call the right sga module, log runtime information, and return the output file name if succeed.
+     * If process erred, the string returned is null.
+     * @param moduleName            SGA module name to be run
+     * @param inputFASTQFile        FASTQ file tobe fed to SGA modul
+     * @param sgaPath               full path to the SGA program
+     * @param workingDir            directory the SGA pipeline is working in
+     * @param threads               threads to use
+     * @param indexer               module representing SGA index
+     * @param indexerArgs           arguments used by SGA index
+     * @param collectiveRuntimeInfo runtime information collected along the process
+     * @return                      the name of file produced by running this SGA module
+     */
+    private static String runAndStopEarly(final String moduleName,
+                                          final File inputFASTQFile,
+                                          final Path sgaPath,
+                                          final File workingDir,
+                                          final int threads,
+                                          final SGAModule indexer,
+                                          final ArrayList<String> indexerArgs,
+                                          final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> collectiveRuntimeInfo){
+
+        SGAUnitRuntimeInfoStruct thisRuntime = null;
+        if(moduleName.equalsIgnoreCase("preprocess")){
+            thisRuntime = runSGAPreprocess(sgaPath, inputFASTQFile, workingDir, indexer, indexerArgs);
+        }else if(moduleName.equalsIgnoreCase("correct")){
+            thisRuntime = runSGACorrect(sgaPath, inputFASTQFile, workingDir, threads, indexer, indexerArgs);
+        }else if(moduleName.equalsIgnoreCase("filter")){
+            thisRuntime = runSGAFilter(sgaPath, inputFASTQFile, workingDir, threads);
+        }else if(moduleName.equalsIgnoreCase("rmdup")){
+            thisRuntime = runSGARmDuplicate(sgaPath, inputFASTQFile, workingDir, threads, indexer, indexerArgs);
+        }else if(moduleName.equalsIgnoreCase("fm-merge")){
+            thisRuntime = runSGAFMMerge(sgaPath, inputFASTQFile, workingDir, threads, indexer, indexerArgs);
+        }else if(moduleName.equalsIgnoreCase("assemble")){
+            thisRuntime = runSGAOverlapAndAssemble(sgaPath, inputFASTQFile, workingDir, threads);
+        }else{
+            throw new GATKException("Wrong module called");
+        }
+
+        collectiveRuntimeInfo.addAll(thisRuntime.runtimeInfo);
+
+        if( thisRuntime.runtimeInfo.stream().mapToInt(entry -> entry._2()._1().ordinal()).sum()!=0 ){
```

Comment:
Might be a little cleaner to do something like

`runtimeInfo.stream().anyMatch(i -> i != ReturnStatus.SUCCESS)`